src/sth_database.js

```diff
@@ -425,6 +425,109 @@
   }
 
   /**
+   * Filters out the results based on the resolution and the optional from and to dates. For certain resolution, the
+   *  from and to dates are considered and applied until the unit of time indicated by the resolution.
+   * @param results The array of results
+   * @param resolution The resolution
+   * @param from The starting date
+   * @param to The ending date
+   * @param aggregatedFunction The aggregation function
+   * @param shouldFilter Flag indicating if null results should be filtered
+   */
+  function filterResults(results, resolution, from, to, aggregatedFunction, shouldFilter) {
+    if (results.length) {
+      var points;
+      if (from &&
+        results[0]._id.origin.getTime() === sthHelper.getOrigin(from, resolution).getTime()) {
+        points = results[0].points;
+        var minOffset;
+        switch (resolution) {
+          case sthConfig.RESOLUTION.SECOND:
+            minOffset = from.getUTCSeconds();
+            break;
+          case sthConfig.RESOLUTION.MINUTE:
+            minOffset = from.getUTCMinutes();
+            break;
+          case sthConfig.RESOLUTION.HOUR:
+            minOffset = from.getUTCHours();
+            break;
+          case sthConfig.RESOLUTION.DAY:
+            minOffset = from.getUTCDate();
+            break;
+          case sthConfig.RESOLUTION.MONTH:
+            minOffset = from.getUTCMonth();
+            break;
+        }
+        for (var i = 0; i < points.length; i++) {
+          if (points[i] && points[i].samples && points[i].offset < minOffset) {
```

Comment:
If I've understood well, this will iterate on a (sorted?) list of points looking for points under the minumum offset in order to "remove" them form the original list. Is it right? In that case, I guess it is not necessary to iterate the whole original list, but you can stop (break) once you et a point over the minumum offset.